### DISCUSSION TITLE
How to properly load my trained model into Python code ?

### DISCUSSION THREAD

**Discussion by huanidz** [PaddlePaddle/PaddleOCR#7377] [Q&A] [answered]
I just finished training a text-recognition with SVTR recognition model for custom dataset, dict. 
The predict result when i run with the command:
python3 tools/infer/predict_rec.py 
--image_dir='/pathToMyImg'
--rec_model_dir='PathToMyInferenceModel' 
--rec_algorithm='SVTR' 
--rec_image_shape='3,64,256' 
--rec_char_dict_path='PathToMyDictFile'
**The output result is correct.**

But when i try to load into python with PaddleOCR(...) from paddleocr module, the prediction now is **very bad**, in my case its repeatly output one of the character in the dict despite the image quality is ok.

And when i try to change to rec_algorithm of PaddleOCR to SVTR, it throw an error: **only supports CRNN, SVTR_LCNet**.

I did check the params for this loader function but changing most of them to fit my model did not help.
I check the dict file, the paths, everything is ok.
The code that load the model run ok but seem like its not.

How do i fix this ?

**Comment by tink2123** [comment, marked as answer]
Currently paddleocr doesn‘t  support the svtr algorithm, and the prediction results will be wrong due to different preprocessing methods. You can try adding SVTR support in paddleocr.

1.  add 'SVTR' in  SUPPORT_REC_MODEL
https://github.com/PaddlePaddle/PaddleOCR/blob/274c216c6771a94807a34fb94377a1d7d674a69f/paddleocr.py#L51

2. rebuild paddleocr
```
python setup.py bdist_wheel
python -m pip install dist/paddleocr-xxx.whl
```

3. predict by your only model

```
paddleocr --det=False --image_dir='/pathToMyImg' --rec_algorithm=SVTR --rec_model_dir='PathToMyInferenceModel' --rec_image_shape='3,64,256' --rec_char_dict_path='PathToMyDictFile'

```
